test/test.go

```diff
@@ -0,0 +1,519 @@
+
+	case "kong.service.request.set_scheme":
+		args := kong_plugin_protocol.String{}
+		e.noErr(proto.Unmarshal(args_d, &args))
+		u, err := url.Parse(e.ServiceReq.Url)
+		e.noErr(err)
+		u.Scheme = args.V
+		e.ServiceReq.Url = u.String()
+
+	case "kong.service.request.set_path":
+		args := kong_plugin_protocol.String{}
+		e.noErr(proto.Unmarshal(args_d, &args))
+		u, err := url.Parse(e.ServiceReq.Url)
+		e.noErr(err)
+		u.Path = args.V
+		e.ServiceReq.Url = u.String()
+
+	case "kong.service.request.set_raw_query":
+		args := kong_plugin_protocol.String{}
+		e.noErr(proto.Unmarshal(args_d, &args))
+		u, err := url.Parse(e.ServiceReq.Url)
+		e.noErr(err)
+		u.RawQuery = args.V
+		e.ServiceReq.Url = u.String()
+
+	case "kong.service.request.set_method":
+		args := kong_plugin_protocol.String{}
+		e.noErr(proto.Unmarshal(args_d, &args))
+		e.ServiceReq.Method = args.V
+
+	case "kong.service.request.set_query":
+		args := structpb.Struct{}
+		e.noErr(proto.Unmarshal(args_d, &args))
+		query := bridge.UnwrapHeaders(&args)
+		u, err := url.Parse(e.ServiceReq.Url)
+		e.noErr(err)
+		u.RawQuery = url.Values(query).Encode()
+		e.ServiceReq.Url = u.String()
+
+	case "kong.service.request.set_header":
+		args := kong_plugin_protocol.KV{}
+		e.noErr(proto.Unmarshal(args_d, &args))
+		e.ServiceReq.Headers.Set(args.K, args.V.GetStringValue())
+
+	case "kong.service.request.add_header":
+		args := kong_plugin_protocol.KV{}
+		e.noErr(proto.Unmarshal(args_d, &args))
+		e.ServiceReq.Headers.Add(args.K, args.V.GetStringValue())
+
+	case "kong.service.request.clear_header":
+		args := kong_plugin_protocol.String{}
+		e.noErr(proto.Unmarshal(args_d, &args))
+		e.ServiceReq.Headers.Del(args.V)
+
+	case "kong.service.request.set_headers":
+		args := structpb.Struct{}
+		e.noErr(proto.Unmarshal(args_d, &args))
+		headers := bridge.UnwrapHeaders(&args)
+		mergeHeaders(e.ServiceReq.Headers, headers)
+
+	case "kong.service.request.set_raw_body":
+		args := kong_plugin_protocol.String{}
+		e.noErr(proto.Unmarshal(args_d, &args))
+		e.ServiceRes.Body = args.V
+
+	case "kong.service.response.get_status":
+		out = &kong_plugin_protocol.Int{V: int32(e.ServiceRes.Status)}
+
+	case "kong.service.response.get_headers":
+		out, err = bridge.WrapHeaders(e.ServiceRes.Headers)
+
+	case "kong.service.response.get_header":
+		args := kong_plugin_protocol.String{}
+		e.noErr(proto.Unmarshal(args_d, &args))
+		out = bridge.WrapString(e.ServiceRes.Headers.Get(args.V))
+
+	case "kong.service.response.get_raw_body":
+		out = bridge.WrapString(e.ServiceRes.Body)
+
+	default:
+		e.t.Errorf("unknown method: \"%v\"", method)
+	}
+
+	if out != nil {
+		e.noErr(err)
+		out_d, err := proto.Marshal(out)
+		e.noErr(err)
+		return out_d
+	}
+	return nil
+}
+
+func (e *testEnv) DoCertificate(config interface{}) {
+	if h, ok := config.(interface{ Certificate(*pdk.PDK) }); ok {
+		e.t.Log("Certificate")
+		h.Certificate(e.pdk)
+	}
+}
+
+func (e *testEnv) DoRewrite(config interface{}) {
```

Comment:
DoRewrite seems are not being called in DoHttp

Reply:
fixed